lib/ancestry/instance_methods.rb

```diff
@@ -18,6 +18,12 @@ def update_descendants_with_new_ancestry
           end
         end
       end
+      unless ancestry_callbacks_disabled?
```

Comment:
So, in my previous implementation:

https://github.com/ManageIQ/manageiq/blob/6b9b4724c6fb63017a44d4917239cf6806698dea/lib/patches/ancestry_patch.rb#L2-L10

The extra `unless` was needed since I needed to call this after `super`.

Now that we are working within a single implementation (and not a monkey patch), we can probably refactor this entire method to share the `if` that is already there:

```ruby
def update_descendants_with_new_ancestry
  # If callbacks enabled
  if !ancestry_callbacks_disabled?
    clear_memoized_instance_variables
    # if node is existing and ancestry was updated and the new ancestry is sane ...
    if !new_record? && ancestry_changed? && sane_ancestry?
      # ... for each descendant ...
      unscoped_descendants.each do |descendant|
        # ... replace old ancestry with new ancestry
        descendant.without_ancestry_callbacks do
          new_ancestor_ids = path_ids + (descendant.ancestor_ids - path_ids_in_database)
          descendant.update_attribute(:ancestor_ids, new_ancestor_ids)
        end
        descendant.clear_memoized_instance_variables
      end
    end
  end
end
```